luxonis_train/nodes/backbones/ghostfacenet.py

```diff
@@ -0,0 +1,511 @@
+            res = self.short_conv(F.avg_pool2d(x, kernel_size=2, stride=2))
+            x1 = self.primary_conv(x)
+            x2 = self.cheap_operation(x1)
+            out = torch.cat([x1, x2], dim=1)
+            return out[:, : self.oup, :, :] * F.interpolate(
+                self.gate_fn(res),
+                size=(out.shape[-2], out.shape[-1]),
+                mode="nearest",
+            )
+
+
+class GhostBottleneckV2(nn.Module):
+    def __init__(
+        self,
+        in_chs,
+        mid_chs,
+        out_chs,
+        dw_kernel_size=3,
+        stride=1,
+        act_layer=nn.PReLU,
+        se_ratio=0.0,
+        layer_id=None,
+        args=None,
+    ):
+        super(GhostBottleneckV2, self).__init__()
+        has_se = se_ratio is not None and se_ratio > 0.0
+        self.stride = stride
+
+        assert layer_id is not None, "Layer ID must be explicitly provided"
+
+        # Point-wise expansion
+        if layer_id <= 1:
+            self.ghost1 = GhostModuleV2(
+                in_chs, mid_chs, prelu=True, mode="original", args=args
+            )
+        else:
+            self.ghost1 = GhostModuleV2(
+                in_chs, mid_chs, prelu=True, mode="attn", args=args
+            )
+
+        # Depth-wise convolution
+        if self.stride > 1:
+            self.conv_dw = nn.Conv2d(
+                mid_chs,
+                mid_chs,
+                dw_kernel_size,
+                stride=stride,
+                padding=(dw_kernel_size - 1) // 2,
+                groups=mid_chs,
+                bias=False,
+            )
+            self.bn_dw = nn.BatchNorm2d(mid_chs)
+
+        # Squeeze-and-excitation
+        if has_se:
+            self.se = SqueezeExcite(mid_chs, se_ratio=se_ratio)
+        else:
+            self.se = None
+
+        self.ghost2 = GhostModuleV2(
+            mid_chs, out_chs, prelu=False, mode="original", args=args
+        )
+
+        # shortcut
+        if in_chs == out_chs and self.stride == 1:
+            self.shortcut = nn.Sequential()
+        else:
+            self.shortcut = nn.Sequential(
+                nn.Conv2d(
+                    in_chs,
+                    in_chs,
+                    dw_kernel_size,
+                    stride=stride,
+                    padding=(dw_kernel_size - 1) // 2,
+                    groups=in_chs,
+                    bias=False,
+                ),
+                nn.BatchNorm2d(in_chs),
+                nn.Conv2d(in_chs, out_chs, 1, stride=1, padding=0, bias=False),
+                nn.BatchNorm2d(out_chs),
+            )
+
+    def forward(self, x):
+        residual = x
+        x = self.ghost1(x)
+        if self.stride > 1:
+            x = self.conv_dw(x)
+            x = self.bn_dw(x)
+        if self.se is not None:
+            x = self.se(x)
+        x = self.ghost2(x)
+        x += self.shortcut(residual)
+        return x
+
+
+# NODES.register_module()
+class GhostFaceNetsV2(BaseNode[torch.Tensor, list[torch.Tensor]]):
```

Comment:
Let's make it `Tensor` instead of `torch.Tensor` for consistency with the rest of the codebase.